packages/plugin-halson-rest-client/src/__tests__/HalsonConfiguratorSpec.js

```diff
@@ -0,0 +1,46 @@
+import HalsonConfigurator from '../HalsonConfigurator';
+
+describe('HalsonConfigurator', () => {
+  const API_ROOT = 'http://localhost/api';
+  const LINK_MAP_RESOLVER = body => body._links;
+
+  it('should fetch the configuration', done => {
+    let httpAgent = {
+      get(url) {
+        expect(url).toBe(API_ROOT + '/');
+        return Promise.resolve({
+          status: 200,
+          body: { _links: { stuff: 'stuff too' } },
+          params: {
+            method: 'GET',
+            url,
+            transformedUrl: url,
+            data: {},
+            headers: {}
+          },
+          headers: {},
+          cached: false
+        });
+      }
+    };
+
+    let configurator = new HalsonConfigurator(
+      httpAgent,
+      API_ROOT,
+      LINK_MAP_RESOLVER
+    );
+    configurator
+      .getConfiguration()
+      .then(configurator => {
+        expect(configurator).toEqual({
+          links: { stuff: 'stuff too' },
+          apiRoot: API_ROOT
+        });
+        done();
+      })
+      .catch(error => {
+        fail(error.stack);
+        done();
```

Comment:
You can call directly `done(error)`, without having to call the `fail` method. See https://jestjs.io/docs/en/asynchronous#callbacks